server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -269,13 +222,27 @@ public SourceFieldMapper build() {
                 }
             }
 
-            SourceFieldMapper sourceFieldMapper = new SourceFieldMapper(
-                sourceMode,
-                enabled.get(),
-                includes.getValue().toArray(Strings.EMPTY_ARRAY),
-                excludes.getValue().toArray(Strings.EMPTY_ARRAY),
-                indexMode
-            );
+            if ((mode.get() == Mode.SYNTHETIC || indexMode == IndexMode.TIME_SERIES)
```

Comment:
I suspect that: `mode.get() == Mode.SYNTHETIC || indexMode == IndexMode.TIME_SERIES` can just be `mode.get() == Mode.SYNTHETIC`.  Since we only should avoid source filtering if source mode is synthetic.

I know this was already the case before. This can be done in a follow up.